docs/crs_compatibility.rst

```diff
@@ -0,0 +1,267 @@
+CRS Compatibility Guide for Geospatial Python
+==============================================
+
+This is meant to be a guide to help you along the way of you use :class:`pyproj.crs.CRS`
+with other Python Geospatial libraries.
+
+.. note:: WKT2 is the best format for storing your CRS according to the
+          `PROJ FAQ <https://proj.org/faq.html#what-is-the-best-format-for-describing-coordinate-reference-systems>`__.
+
+
+osgeo/gdal
+----------
+
+https://github.com/osgeo/gdal
+
+Converting from `osgeo.osr.SpatialReference` to `pyproj.crs.CRS`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    from osgeo.osr import SpatialReference
+    from pyproj.crs import CRS
```

Comment:
I think most other examples in the docs do `from pyproj import CRS`, might be good to be somewhat consistent

Reply:
I am a big fan of consistency! Here is why some things are a little inconsistent.

The access pattern `from pyproj import CRS` was to be consistent with how things were done in `pyproj` 1 as the `Proj` and `Geod` classes were defined in the `pyproj.__init__.py`. It is also simplifies things when using the `Transformer` class as you only have to do one import `from pyproj import CRS, Transformer`. This is a common access pattern and so it is nice to simplify things a bit there. It is also nice to not have to type an extra 4 characters to import the `CRS` class in the terminal :).

The access pattern `from pyproj.crs import CRS` is consistent with the community `CRS` access patterns (`rasterio.crs` and `cartopy.crs`), which is why I chose to use it in this document. It is also nice to use this pattern as a one line import when accessing the other CRS objects such as `Datum` (https://github.com/pyproj4/pyproj/pull/596#issuecomment-614448662). For example, `from pyproj.crs import CRS, Datum`.

I like using them both personally and they both have their uses. They are both consistent within their own realm of usage. However, they can be inconsistent with each other. :man_shrugging:

Reply:
My opinionated(!) take:

> The access pattern from pyproj.crs import CRS is consistent with the community CRS access patterns

That are packages that are about much more than a CRS, so for them it makes more sense to put it in a submodule, than it does for pyproj, IMO. The CRS is the core object users of pyproj should be using, so I think ideally that is just top-level.

> which is why I chose to use it in this document.

Especially for this guide, with all the different CRS objects coming from different packages, I think doing a `import pyproj` and `pyproj.CRS` might actually be more readable. Because then, it is directly clear that it is about the pyproj CRS object, without needing to check the imports